### ISSUE TITLE
Alpha transparency colors : some are wrong

### ISSUE THREAD

**Issue by Maeda1** [jaor/xmobar#246]
Hi !

I just set up alpha transparency, which works nicely, but some colors are different when rendering with alpha.
Those from blue/red/green colors are ok, but if I choose a orange color tint as along as alpha transparency (other than value = 255), it goes rather purple ! Same results with yellow based colors.

What do you think about that ?

To reproduce the problem : 
- choose for example : `bgColor = "#FFAC4B"`
- choose `alpha = [value other than 0 and 255]`

=> With and without alpha are two different colors...

My config : 
xmobar 0.23.1-5
(+ xmonad 0.11.1-5)

**Comment by jaor** [comment]
what's the color of the background on top of which xmobar is? i think
we're merging with that too...

**Comment by Maeda1** [comment]
I tried with a full black background and a full white background, the tint is the same, rather pink (far away from orange). It's not related to the background I think.

**Comment by Maeda1** [comment]
Do you manage to get the bug as described ?

**Comment by jaor** [comment]
@Maeda1 sorry, i've been swamped with work and real life chores... i'll try to check during the following days.

**Comment by Maeda1** [comment]
That's ok, it's not critical bug ;)

**Comment by Maeda1** [comment]
Some new informations : 

Same with xmonad 0.12-2.
Other color that doesn't show as expected : 
`bgColor = "#196519"`
It should be rather green, but it's completely blue, and doesn't change with another value of alpha.

**Comment by jaor** [comment]
@Maeda1 i've at least reproduced what you see... more or less... i see pink and purple, and, as you say, that doesn't seem right.  there's probably something with the way we mask the color to produce the transparency, but i don't know yet what it is.

**Comment by Maeda1** [comment]
OK, glad to hear that you have it too. The strangest thing is when the problem occurs, all the value of alpha in range [1-254] doesn't seems to affect the overall transparency, maybe it's a clue to investigate.
I think I'm not skilled enough to debug it myself :(

By the way, thanks for testing it.

**Comment by jaor** [comment]
hmm, in my case, the alpha range does have an effect in the overall transparency...

**Comment by Maeda1** [comment]
Oh yeah me too.
Forget what I said at my previous post, I was wrong.
The biggest difference is between alpha with value 254 and 255.

**Comment by Maeda1** [comment]
I have a question regardless of this bug (do I need to open another topic ?) : who decided to set the alpha transparency like that : 

> "The transparency: 0 is transparent, 255 is opaque. Default: 255"

I mean, I think people using xmobar do use trayer, and I have to do a 255 minus the value of xmobarrc's alpha to have the same transparency.

Do you think inverting the opaque to 0 and the full tranparency 255 is possible ?

Here is what I do to get the same transparency between xmobar and trayer : 
`trayer --edge top --align right --widthtype request --expand true --SetDockType true --SetPartialStrut true --transparent true --alpha **$(echo "`expr 255 - $(cat ~/.xmobar/xmobarrc | grep alpha | cut -d "=" -f2 | cut -d " " -f2)`")** ... &`

**Comment by jaor** [comment]
i'm a bit worried changing that parameter will mess up with other
people's setups... we could perhaps introduce a new one, that is opaque
for 0 and transparent for 255, although i'm having a hard time trying to
come up with a good name: suggestions?

**Comment by Maeda1** [comment]
No you're right, let's keep that. If there is someone using trayer with xmobar like me, he'll find the command to match xmobar alpha's value to trayer alpha's value, on my previous posts.

**Comment by seanlaguna** [comment]
For me alpha works fine but the G and B channels of the bgcolor get switched when I use non-255 alpha!

**Comment by Maeda1** [comment]
Maybe I get the same as yours, and the cause is this switch. Did you try the value I wrote some post ago, and get pink tint ?

**Comment by Maeda1** [comment]
seanlaguna: you mean that if using a hex color code with non-alpha transparency and using the same hex code, but switching G with B value, with alpha transparency = the tint is good ?

**Comment by seanlaguna** [comment]
Yes

**Comment by Maeda1** [comment]
Indeed, it works :). That's a workaround, but where is located the bug ?

**Comment by Maeda1** [comment]
Maybe this workaround is enough to not keep this bug opened ?
And you should put an information in xmobar man when using this option.

Good idea ?

**Comment by jaor** [comment]
@Maeda1 yes, might be a good idea, specially since i cannot find a reason in the code of why this is happening.  at the very least, documenting it will work as a stop-gap until a real fix is available.  added to my TODO!

**Comment by sebasijan** [comment]
Is there any update on this? Whenever I set `alpha` value on my xmobar which is not 255 it simply turns solid black

**Comment by a1eaiactaest** [comment]
Works as expected for me. I can set any value between 0 and 255.

**Comment by Maeda1** [comment]
Hello,
Recently updated my whole system (Archlinux 64 bits), and got xmobar 0.39-122.

It seems we do not need to invert G and B as advised by @seankhl.
Example : wanting a (alpha enabled) background color in green : I can now use bgColor = "#20**275e**" and not anymore bgColor = "#20**5e27**"

Expected color green is OK with the "true" code.

Any input on why that happened (which is good)?

**Comment by liskin** [comment]
@Maeda1 likely related to https://github.com/xmonad/X11-xft/commit/0f0066a71eef5ac8d879c84a42f0622a38cc12fa which got released in X11-xft-0.3.3

**Comment by Maeda1** [comment]
@liskin: very likely related to. Thanks!
